### PR TITLE
Settings functionality

### PR DESCRIPTION
**Pull request by YAWAL**
Functionality for settings has been completely rewritten according to changes on server as we discussed today

### CODE REVIEW

File: settingService/blockUser.go [hunk 1]
```diff
@@ -78,17 +77,11 @@ type ContactResult struct {
...
 // IsUserBlocked checks if user is blocked for chatting in contacts table
-// Function returns: if succeed - is_blocked value(0 or 1) from contacts table and nil, if failed - nil and error
-func IsUserBlocked(byteRequest [] byte) (isBlocked int, err error) {
-	request, err := messageService.UnmarshalMessage(byteRequest)
-	if err != nil {
-		loger.Log.Errorf("Error has occurred: ", err)
-		return nil, err
-	}
+// Function returns: if succeed - true or false from contacts table(depend is contact blocked or not) and nil, if failed - nil and error
+func IsUserBlocked(request *messageService.Message) (isBlocked bool, err error) {
```
Comment (8tomat8) [inline settingService/blockUser.go:81]: You defined `isBlocked` and `err`, but didn't use them

File: settingService/blockUser.go [hunk 2]
```diff
@@ -106,12 +99,11 @@ func IsUserBlocked(byteRequest [] byte) (isBlocked int, err error) {
...
+	if result.IsBlocked == 0 {
+		return false, nil
 	}
-	if isBlocked == 1 {
-		return 1, nil
+	if result.IsBlocked == 1 {
+		return true, nil
 	}
 	err = errors.New("Function has failed")
 	return nil, err
```
Comment (8tomat8) [inline settingService/blockUser.go]: How `nil` could be returned as type `bool`?

File: server/modules/modules.go [hunk 3]
```diff
@@ -48,5 +49,96 @@ func Auth(message *messageService.Message, conn *websocket.Conn) {
...
+		loger.Log.Warnf(" Info about user has not changed. Please try again")
+		byteOk, _ := json.Marshal(changeUserInfoOk) // this for UI
+		conn.WriteMessage(messageType, byteOk)      // this for UI
+		return
+	}
+	return
+}
+
+func BlockUnblockUser(message *messageService.Message, messageType int, conn *websocket.Conn) {
+	ok, err := settingService.BlockUnblockUser(*message)
```
Comment (8tomat8) [inline server/modules/modules.go]: Why are you trying to copy value to function if it require pointer? Have you tried to run this code?

**Review by 8tomat8 — CHANGES_REQUESTED** [review]

File: settingService/blockUser.go (hunk 2 above)
Comment (YAWAL) [reply settingService/blockUser.go]: Fixed. In a case of failed function will return true, which means that message mustn't be sent.

File: server/modules/modules.go (hunk 3 above)
Comment (YAWAL) [reply server/modules/modules.go]: Yes, functions should accept pointer. A problem has been fixed.

File: settingService/blockUser.go (hunk 1 above)
Comment (YAWAL) [reply settingService/blockUser.go:81]: Checking for blocking has been added into Message function. Now, contact will be checked before showing a message for a user.

**Review by 8tomat8 — APPROVED** [review]

File: settingService/blockUser.go [hunk 4]
```diff
@@ -78,20 +77,15 @@ type ContactResult struct {
...
-func IsUserBlocked(byteRequest [] byte) (isBlocked int, err error) {
-	request, err := messageService.UnmarshalMessage(byteRequest)
-	if err != nil {
-		loger.Log.Errorf("Error has occurred: ", err)
-		return nil, err
-	}
+// Function returns: if succeed - true or false from contacts table(depend is contact blocked or not) and nil, if failed - nil and error
+func IsUserBlocked(request *messageService.Message) (isBlocked bool, err error) {
+	isBlocked = true
 	contactUser := request.Body.ReceiverName
```
Comment (justakit) [inline settingService/blockUser.go:83]: Unresolved reference 'ReceiverName'
Body has type `RawMessage`. So you cannot get `ReceiverName` in this way

File: server/modules/modules.go [hunk 5]
```diff
@@ -17,7 +20,19 @@ func EmptyType() {
...
+		byteError, _ := json.Marshal(changeNicknameError) // this for UI
+		conn.WriteMessage(messageType, byteError)         // this for UI
+		return
+	}
+	if isBlocked {
+		loger.Log.Errorf("This contact is blocked")
+		return
+	}
+
+	err = coremessage.SendMessage(message, messageType)
```
Comment (justakit) [inline server/modules/modules.go:35]: Cannot use `coremessage.SendMessage(message, messageType) (type messageService.Message)` as type `error` in assignment

**Pull request merged**